application/src/main/java/org/togetherjava/tjbot/features/github/GitHubLinkCommand.java

```diff
@@ -0,0 +1,119 @@
+     * @param config the config to get the GitHub API key
+     */
+    public GitHubLinkCommand(Database database, Config config) {
+        super("link-gh-project",
+                "Links a GitHub repository to this project post to receive pull request notifications",
+                CommandVisibility.GUILD);
+        this.database = database;
+        this.githubApiKey = config.getGitHubApiKey();
+
+        getData()
+            .addOption(OptionType.STRING, REPOSITORY_OWNER_OPTION,
+                    "The owner of the repository to be linked", true)
+            .addOption(OptionType.STRING, REPOSITORY_NAME_OPTION,
+                    "The name of the repository to be linked", true);
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        OptionMapping repositoryOwnerOption = event.getOption(REPOSITORY_OWNER_OPTION);
+        OptionMapping repositoryNameOption = event.getOption(REPOSITORY_NAME_OPTION);
+
+        if (repositoryOwnerOption == null || repositoryNameOption == null) {
+            event.reply("You must specify a repository owner and a repository name")
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        long channelId = event.getChannelIdLong();
+        String repositoryOwner = repositoryOwnerOption.getAsString();
+        String repositoryName = repositoryNameOption.getAsString();
+
+        GitHub github;
+        try {
+            github = new GitHubBuilder().withOAuthToken(githubApiKey).build();
+        } catch (IOException e) {
+            logger.error("Failed to initialize GitHub API wrapper.", e);
+            event.reply("Failed to initialize GitHub API wrapper.").setEphemeral(true).queue();
+            return;
+        }
+
+        try {
+            if (!isRepositoryAccessible(github, repositoryOwner, repositoryName)) {
+                event.reply("Repository is not publicly available.").setEphemeral(true).queue();
+                logger.info("Repository {}/{} is not accessible.", repositoryOwner, repositoryName);
+                return;
+            }
+        } catch (IOException e) {
+            logger.error("Failed to check if GitHub repository is available.", e);
+            event.reply("Failed to link repository.").setEphemeral(true).queue();
+            return;
+        }
+
+        try {
+            saveNotificationToDatabase(channelId, repositoryOwner, repositoryName);
+            event.reply("Successfully linked repository.").setEphemeral(true).queue();
+        } catch (DatabaseException e) {
+            logger.error("Failed to save pull request notification to database.", e);
+            event.reply("Failed to link repository.").setEphemeral(true).queue();
+        }
+    }
+
+    private boolean isRepositoryAccessible(GitHub github, String owner, String name)
+            throws IOException {
+        GHRepository repository = github.getRepository(owner + "/" + name);
```

Comment:
this can be static and also just `return github.getRepository(owner + "/" + name) != null;`